### ISSUE TITLE
patch only update

### ISSUE THREAD

**Issue by sanchek** [raineorshine/npm-check-updates#477]
Hello, is it possible to update only patch(~), not latest version?

* node version: 8.14.0
* npm version: 6.4.1
* npm-check-updates version: 2.15.0

---------------------------

**Comment by raineorshine** [comment]
Not currently, but the `--semverLevel` option could be extended to support this.

Can you describe the use case? Specifically, why not just use version ranges such as `1.0.x`?

**Comment by raineorshine** [comment]
Can you use version ranges like `1.0.x` in your package.json and then simply use `npm update`?

**Comment by stoically** [comment]
Maybe I'm missing something here, but `--semverLevel minor` should already update only patch versions. The naming is a bit misleading, but I assume it's meant like `major` keeps the major version and updates minor+patch, `minor` keeps the major/minor version and updates the patch. Note however that `--semverLevel` only supports direct versions in the package.json, no version ranges (related: #492).

**Comment by matushorvath** [comment]
@stoically The documentation suggests it should work like you say, but it does not actually work. For example:

```
$ ncu
Checking [censored]/package.json
[====================] 14/14 100%

 @sbn/logger           ^2.0.18  →    ^2.0.21
 @sbn/dts-tools        ^1.0.41  →    ^1.0.42
 @sbn/tslint-config   ^2.22.33  →   ^2.22.34
 @types/jest           ^24.0.0  →   ^24.0.11
 @types/node          ^10.14.1  →   ^11.11.5
 @types/sbn__logger    ^2.0.18  →    ^2.0.21
 ts-jest              ^23.10.5  →    ^24.0.0
 ts-node                ^7.0.1  →     ^8.0.3
 typedoc               ^0.12.0  →    ^0.14.2
 typescript          ^3.3.3333  →  ^3.3.4000

Run ncu -u to upgrade package.json
$ ncu --semverLevel major
Checking [censored]/package.json
[====================] 14/14 100%

All dependencies match the major package versions :)
$ ncu --semverLevel minor
Checking [censored]/package.json
[====================] 14/14 100%

All dependencies match the minor package versions :)
```

I would have expected the second call to also update the packages, with the difference that @types/node would be upgraded to 10.x latest version (not 11.x), ts-jest to 23.x latest version (not 24.x) and ts-node to 7.x latest version (not 8.x). And in the last case I would expect some updates, but only the last version number (patch) would change.

**Comment by stoically** [comment]
@matushorvath That's because those are all *version ranges*, not direct versions: "Note however that --semverLevel only supports direct versions in the package.json, no version ranges". To support ranges someone would need to implement it in ncu, as noted in #492.

**Comment by matushorvath** [comment]
@stoically Oh I see. I did not realize that version X and up is also a version range, but it makes sense. Thanks for the explanation.

**Comment by fireairforce** [comment]
@stoically Thx for your explanation, i meet the same question...

**Comment by raineorshine** [comment]
Added `--semverLevel` support for version ranges. Published in `v6.0.0`.

```
ncu --semverLevel minor
```
